test/Microsoft.AspNetCore.Routing.Tests/Template/TemplateParserTests.cs

```diff
@@ -554,8 +554,8 @@ public void Parse_ComplexSegment_OptionalParametersSeperatedByPeriod_Invalid(str
             // Act and Assert
             ExceptionAssert.Throws<ArgumentException>(
                 () => TemplateParser.Parse(template),
-                "In the complex segment '"+ template +"',  the optional parameter 'p2' is preceded by an invalid " + 
-                "segment '" + parameter +"'. Only valid literal to precede an optional parameter is a period (.)." + 
+                "In the complex segment '"+ template +"',  the optional parameter 'p2' is preceded by an invalid " +
+                "segment '" + parameter +"'. Only valid literal to precede an optional parameter is a period (.)." +
```

Comment:
Argh, since we're here, can we fix the grammar here too? `**The** only valid literal **that can** precede an...`